src/hooks/useENSRegistrationActionHandler.ts

```diff
@@ -0,0 +1,265 @@
+import { updateTransactionRegistrationParameters } from '@rainbow-me/redux/ensRegistration';
+import { timeUnits } from '@rainbow-me/references';
+
+enum REGISTRATION_STEPS {
+  COMMIT = 'COMMIT',
+  WAIT_COMMIT_CONFIRMATION = 'WAIT_COMMIT_CONFIRMATION',
+  WAIT_ENS_COMMITMENT = 'WAIT_ENS_COMMITMENT',
+  REGISTER = 'REGISTER',
+}
+
+const ENS_SECONDS_WAIT = 60;
+
+export default function useENSRegistrationActionHandler(
+  {
+    yearsDuration,
+  }: {
+    yearsDuration: number;
+  } = {} as any
+) {
+  const dispatch = useDispatch();
+  const { accountAddress, network } = useAccountSettings();
+  const getNextNonce = useCurrentNonce(accountAddress, network);
+  const { registrationParameters } = useENSProfile();
+  const [stepGasLimit, setStepGasLimit] = useState<string | null>(null);
+  const [
+    secondsSinceCommitConfirmed,
+    setSecondsSinceCommitConfirmed,
+  ] = useState(
+    registrationParameters?.commitTransactionConfirmedAt
+      ? differenceInSeconds(
+          Date.now(),
+          registrationParameters.commitTransactionConfirmedAt
+        )
+      : -1
+  );
+
+  const commit = useCallback(
+    async (callback: () => void) => {
+      const { name } = registrationParameters as RegistrationParameters;
+      const wallet = await loadWallet();
+      if (!wallet) {
+        return;
+      }
+
+      const salt = generateSalt();
+      const nonce = await getNextNonce();
+      const rentPrice = await getRentPrice(
+        name,
+        yearsDuration * timeUnits.secs.year
+      );
+
+      const commitEnsRegistrationParameters: ENSActionParameters = {
+        duration: yearsDuration * timeUnits.secs.year,
+        name,
+        nonce,
+        ownerAddress: accountAddress,
+        rentPrice: rentPrice.toString(),
+        salt,
+      };
+
+      await executeRap(
+        wallet,
+        RapActionTypes.commitENS,
+        commitEnsRegistrationParameters,
+        callback
+      );
+    },
+    [accountAddress, yearsDuration, getNextNonce, registrationParameters]
+  );
+
+  const register = useCallback(
+    async (callback: () => void) => {
+      const {
+        name,
+        duration,
+        records,
+        salt,
+      } = registrationParameters as RegistrationParameters;
+      const wallet = await loadWallet();
+      if (!wallet) {
+        return;
+      }
+
+      const nonce = await getNextNonce();
+      const rentPrice = await getRentPrice(name, duration);
+
+      const registerEnsRegistrationParameters: ENSActionParameters = {
+        duration,
+        name,
+        nonce,
+        ownerAddress: accountAddress,
+        records,
+        rentPrice: rentPrice.toString(),
+        salt,
+      };
+
+      await executeRap(
+        wallet,
+        RapActionTypes.registerENS,
+        registerEnsRegistrationParameters,
+        callback
+      );
+    },
+    [accountAddress, getNextNonce, registrationParameters]
+  );
+
+  const registrationStep = useMemo(() => {
```

Comment:
the idea of this is that the UI doesn't have to pay attention to the state whatsoever, it just needs to get the `action`, `step` (to modify the UI accordingly), and the `gasLimit` of the step